src/main/kotlin/eu/europa/ec/eudi/pidissuer/adapter/out/diploma/IssueJwtVcJsonDiploma.kt

```diff
@@ -0,0 +1,188 @@
+import eu.europa.ec.eudi.pidissuer.port.out.IssueSpecificCredential
+import eu.europa.ec.eudi.pidissuer.port.out.persistence.GenerateNotificationId
+import eu.europa.ec.eudi.pidissuer.port.out.persistence.StoreIssuedCredential
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import kotlinx.serialization.json.*
+import org.slf4j.LoggerFactory
+import java.net.URI
+import java.time.Clock
+import java.util.Locale.ENGLISH
+
+val DiplomaJwtVcJsonScope: Scope = Scope("diploma_jwt_vc_json")
+
+val diplomaDisplay = listOf(
+    CredentialDisplay(
+        name = DisplayName("University Degree", ENGLISH),
+        logo = ImageUri(
+            uri = URI.create("https://examplestate.com/public/diploma.png"),
+            alternativeText = "A square figure of a Diploma",
+        ),
+    ),
+)
+
+val IssueJwtVcJsonDiplomaConfiguration = JwtVcJsonCredentialConfiguration(
+    id = CredentialConfigurationId((DiplomaJwtVcJsonScope.value)),
+    scope = DiplomaJwtVcJsonScope,
+    cryptographicBindingMethodsSupported = emptySet(),
+    credentialSigningAlgorithmsSupported = emptySet(),
+    display = diplomaDisplay,
+    proofTypesSupported = nonEmptySetOf(ProofType.Jwt(nonEmptySetOf(JWSAlgorithm.RS256, JWSAlgorithm.ES256))),
+    credentialDefinition = CredentialDefinition(
+        type = listOf("VerifiableCredential", "UniversityDegreeCredential"),
+        credentialSubject = listOf(
+            AchievedClaim.attribute,
+            EntitledToClaim.attribute,
+            PerformedClaim.attribute,
+        ),
+    ),
+)
+
+class IssueJwtVcJsonDiploma(
+    val credentialIssuerId: CredentialIssuerId,
+    private val generateUserDiplomaData: GenerateUserDiplomaData,
+    private val notificationsEnabled: Boolean,
+    private val generateNotificationId: GenerateNotificationId,
+    private val extractJwkFromCredentialKey: ExtractJwkFromCredentialKey,
+    private val issuerSigningKey: IssuerSigningKey,
+    private val clock: Clock,
+    private val storeIssuedCredential: StoreIssuedCredential,
+) : IssueSpecificCredential<JsonElement> {
+
+    private val log = LoggerFactory.getLogger(IssueJwtVcJsonDiploma::class.java)
+
+    override val supportedCredential: JwtVcJsonCredentialConfiguration
+        get() = IssueJwtVcJsonDiplomaConfiguration
+
+    override val publicKey: JWK? = null
+
+    private val validateProof = ValidateProof(credentialIssuerId)
+
+    context(Raise<IssueCredentialError>)
+    override suspend fun invoke(
+        authorizationContext: AuthorizationContext,
+        request: CredentialRequest,
+        credentialIdentifier: CredentialIdentifier?,
+        expectedCNonce: CNonce,
+    ): CredentialResponse<JsonElement> = coroutineScope {
+        log.info("Handling issuance request ...")
+        val holderPubKey = async(Dispatchers.Default) { holderPubKey(request, expectedCNonce) }
+
+        val jwt = encodeDiplomaToJwtVcJson(holderPubKey.await())
```

Comment:
lookup the user diploma (generate it), then pass it to `encodeDiplomaToJwtVCJson(diploma, holderPubKey)`